### ISSUE TITLE
Search by state ID and Country ID

### ISSUE THREAD

**Issue by cachly** [cachly/bug-tracker#1078]
**[Original report](https://bitbucket.org/zedsaidstudio/cachly/issues/1078) by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

**Comment by cachly** [comment]
**Original comment by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

```html
<select id="state_id2" name="state_id" class="Select EqualWidthSelect">
  <option value="60">Alabama</option>
  <option value="2">Alaska</option>
  <option value="3">Arizona</option>
  <option value="4">Arkansas</option>
  <option value="5">California</option>
  <option value="6">Colorado</option>
  <option value="7">Connecticut</option>
  <option value="9">Delaware</option>
  <option value="8">District of Columbia</option>
  <option value="10">Florida</option>
  <option value="11">Georgia</option>
  <option value="12">Hawaii</option>
  <option value="13">Idaho</option>
  <option value="14">Illinois</option>
  <option value="15">Indiana</option>
  <option value="16">Iowa</option>
  <option value="17">Kansas</option>
  <option value="18">Kentucky</option>
  <option value="19">Louisiana</option>
  <option value="20">Maine</option>
  <option value="21">Maryland</option>
  <option value="22">Massachusetts</option>
  <option value="23">Michigan</option>
  <option value="24">Minnesota</option>
  <option value="25">Mississippi</option>
  <option value="26">Missouri</option>
  <option value="27">Montana</option>
  <option value="28">Nebraska</option>
  <option value="29">Nevada</option>
  <option value="30">New Hampshire</option>
  <option value="31">New Jersey</option>
  <option value="32">New Mexico</option>
  <option value="33">New York</option>
  <option value="34">North Carolina</option>
  <option value="35">North Dakota</option>
  <option value="36">Ohio</option>
  <option value="37">Oklahoma</option>
  <option value="38">Oregon</option>
  <option value="39">Pennsylvania</option>
  <option value="40">Rhode Island</option>
  <option value="41">South Carolina</option>
  <option value="42">South Dakota</option>
  <option value="43">Tennessee</option>
  <option value="44">Texas</option>
  <option value="45">Utah</option>
  <option value="46">Vermont</option>
  <option value="47">Virginia</option>
  <option value="48">Washington</option>
  <option value="49">West Virginia</option>
  <option value="50">Wisconsin</option>
  <option value="51">Wyoming</option>
</select>
```

**Comment by cachly** [comment]
**Original comment by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

Countries

```html
<select id="by_country" name="country_id" class="Select EqualWidthSelect">
  <option value="12">Afghanistan</option>
  <option value="272">Aland Islands</option>
  <option value="244">Albania</option>
  <option value="14">Algeria</option>
  <option value="245">American Samoa</option>
  <option value="16">Andorra</option>
  <option value="17">Angola</option>
  <option value="246">Anguilla</option>
  <option value="18">Antarctica</option>
  <option value="13">Antigua and Barbuda</option>
  <option value="19">Argentina</option>
  <option value="15">Armenia</option>
  <option value="20">Aruba</option>
  <option value="3">Australia</option>
  <option value="227">Austria</option>
  <option value="21">Azerbaijan</option>
  <option value="23">Bahamas</option>
  <option value="29">Bahrain</option>
  <option value="24">Bangladesh</option>
  <option value="25">Barbados</option>
  <option value="40">Belarus</option>
  <option value="4">Belgium</option>
  <option value="31">Belize</option>
  <option value="26">Benin</option>
  <option value="27">Bermuda</option>
  <option value="30">Bhutan</option>
  <option value="32">Bolivia</option>
  <option value="275">Bonaire</option>
  <option value="234">Bosnia and Herzegovina</option>
  <option value="33">Botswana</option>
  <option value="247">Bouvet Island</option>
  <option value="34">Brazil</option>
  <option value="248">British Indian Ocean Territories</option>
  <option value="39">British Virgin Islands</option>
  <option value="36">Brunei</option>
  <option value="37">Bulgaria</option>
  <option value="216">Burkina Faso</option>
  <option value="35">Burundi</option>
  <option value="42">Cambodia</option>
  <option value="43">Cameroon</option>
  <option value="5">Canada</option>
  <option value="239">Cape Verde</option>
  <option value="44">Cayman Islands</option>
  <option value="46">Central African Republic</option>
  <option value="249">Chad</option>
  <option value="6">Chile</option>
  <option value="47">China</option>
  <option value="250">Christmas Island</option>
  <option value="251">Cocos (Keeling) Islands</option>
  <option value="49">Colombia</option>
  <option value="50">Comoros</option>
  <option value="51">Congo</option>
  <option value="48">Cook Islands</option>
  <option value="52">Costa Rica</option>
  <option value="53">Croatia</option>
  <option value="238">Cuba</option>
  <option value="54">Curacao</option>
  <option value="55">Cyprus</option>
  <option value="56">Czech Republic</option>
  <option value="257">Democratic Republic of the Congo</option>
  <option value="57">Denmark</option>
  <option value="58">Djibouti</option>
  <option value="59">Dominica</option>
  <option value="60">Dominican Republic</option>
  <option value="252">East Timor</option>
  <option value="61">Ecuador</option>
  <option value="63">Egypt</option>
  <option value="64">El Salvador</option>
  <option value="62">Equatorial Guinea</option>
  <option value="65">Eritrea</option>
  <option value="66">Estonia</option>
  <option value="67">Ethiopia</option>
  <option value="69">Falkland Islands</option>
  <option value="68">Faroe Islands</option>
  <option value="71">Fiji</option>
  <option value="72">Finland</option>
  <option value="73">France</option>
  <option value="70">French Guiana</option>
  <option value="74">French Polynesia</option>
  <option value="253">French Southern Territories</option>
  <option value="75">Gabon</option>
  <option value="76">Gambia</option>
  <option value="78">Georgia</option>
  <option value="79">Germany</option>
  <option value="254">Ghana</option>
  <option value="80">Gibraltar</option>
  <option value="82">Greece</option>
  <option value="83">Greenland</option>
  <option value="81">Grenada</option>
  <option value="77">Guadeloupe</option>
  <option value="229">Guam</option>
  <option value="84">Guatemala</option>
  <option value="86">Guernsey</option>
  <option value="255">Guinea</option>
  <option value="85">Guinea-Bissau</option>
  <option value="87">Guyana</option>
  <option value="89">Haiti</option>
  <option value="256">Heard Island And Mcdonald Islands</option>
  <option value="90">Honduras</option>
  <option value="91">Hong Kong</option>
  <option value="92">Hungary</option>
  <option value="93">Iceland</option>
  <option value="94">India</option>
  <option value="95">Indonesia</option>
  <option value="96">Iran</option>
  <option value="97">Iraq</option>
  <option value="7">Ireland</option>
  <option value="243">Isle of Man</option>
  <option value="98">Israel</option>
  <option value="99">Italy</option>
  <option value="100">Ivory Coast</option>
  <option value="101">Jamaica</option>
  <option value="104">Japan</option>
  <option value="102">Jersey</option>
  <option value="103">Jordan</option>
  <option value="106">Kazakhstan</option>
  <option value="107">Kenya</option>
  <option value="109">Kiribati</option>
  <option value="241">Kuwait</option>
  <option value="108">Kyrgyzstan</option>
  <option value="110">Laos</option>
  <option value="111">Latvia</option>
  <option value="113">Lebanon</option>
  <option value="114">Lesotho</option>
  <option value="115">Liberia</option>
  <option value="112">Libya</option>
  <option value="116">Liechtenstein</option>
  <option value="117">Lithuania</option>
  <option value="8">Luxembourg</option>
  <option value="258">Macau</option>
  <option value="125">Macedonia</option>
  <option value="119">Madagascar</option>
  <option value="129">Malawi</option>
  <option value="121">Malaysia</option>
  <option value="124">Maldives</option>
  <option value="127">Mali</option>
  <option value="128">Malta</option>
  <option value="240">Marshall Islands</option>
  <option value="122">Martinique</option>
  <option value="123">Mauritania</option>
  <option value="134">Mauritius</option>
  <option value="259">Mayotte</option>
  <option value="228">Mexico</option>
  <option value="242">Micronesia</option>
  <option value="237">Moldova</option>
  <option value="130">Monaco</option>
  <option value="131">Mongolia</option>
  <option value="274">Montenegro</option>
  <option value="135">Montserrat</option>
  <option value="132">Morocco</option>
  <option value="133">Mozambique</option>
  <option value="136">Myanmar</option>
  <option value="137">Namibia</option>
  <option value="138">Nauru</option>
  <option value="140">Nepal</option>
  <option value="141">Netherlands</option>
  <option value="41">New Caledonia</option>
  <option value="9">New Zealand</option>
  <option value="144">Nicaragua</option>
  <option value="143">Niger</option>
  <option value="145">Nigeria</option>
  <option value="149">Niue</option>
  <option value="260">Norfolk Island</option>
  <option value="236">Northern Mariana Islands</option>
  <option value="147">Norway</option>
  <option value="150">Oman</option>
  <option value="151">Pakistan</option>
  <option value="261">Palau</option>
  <option value="276">Palestine</option>
  <option value="152">Panama</option>
  <option value="156">Papua New Guinea</option>
  <option value="262">Paraguay</option>
  <option value="153">Peru</option>
  <option value="154">Philippines</option>
  <option value="155">Pitcairn Islands</option>
  <option value="158">Poland</option>
  <option value="159">Portugal</option>
  <option value="226">Puerto Rico</option>
  <option value="160">Qatar</option>
  <option value="161">Reunion</option>
  <option value="162">Romania</option>
  <option value="163">Russia</option>
  <option value="164">Rwanda</option>
  <option value="277">Saba</option>
  <option value="171">Saint Helena</option>
  <option value="264">Saint Kitts and Nevis</option>
  <option value="173">Saint Lucia</option>
  <option value="217">Samoa</option>
  <option value="183">San Marino</option>
  <option value="176">Sao Tome and Principe</option>
  <option value="166">Saudi Arabia</option>
  <option value="167">Senegal</option>
  <option value="222">Serbia</option>
  <option value="168">Seychelles</option>
  <option value="178">Sierra Leone</option>
  <option value="179">Singapore</option>
  <option value="182">Slovakia</option>
  <option value="181">Slovenia</option>
  <option value="184">Solomon Islands</option>
  <option value="185">Somalia</option>
  <option value="165">South Africa</option>
  <option value="267">South Georgia and Sandwich Islands</option>
  <option value="180">South Korea</option>
  <option value="278">South Sudan</option>
  <option value="186">Spain</option>
  <option value="187">Sri Lanka</option>
  <option value="169">St Barthelemy</option>
  <option value="170">St Eustatius</option>
  <option value="175">St Pierre Miquelon</option>
  <option value="177">St Vince Grenadines</option>
  <option value="174">St. Martin</option>
  <option value="188">Sudan</option>
  <option value="189">Suriname</option>
  <option value="268">Svalbard and Jan Mayen</option>
  <option value="190">Swaziland</option>
  <option value="10">Sweden</option>
  <option value="192">Switzerland</option>
  <option value="193">Syria</option>
  <option value="194">Taiwan</option>
  <option value="195">Tajikistan</option>
  <option value="196">Tanzania</option>
  <option value="198">Thailand</option>
  <option value="200">Togo</option>
  <option value="269">Tokelau</option>
  <option value="201">Tonga</option>
  <option value="202">Trinidad and Tobago</option>
  <option value="203">Tunisia</option>
  <option value="204">Turkey</option>
  <option value="199">Turkmenistan</option>
  <option value="197">Turks and Caicos Islands</option>
  <option value="205">Tuvalu</option>
  <option value="208">Uganda</option>
  <option value="207">Ukraine</option>
  <option value="206">United Arab Emirates</option>
  <option value="11">United Kingdom</option>
  <option value="210">Uruguay</option>
  <option value="270">US Minor Outlying Islands</option>
  <option value="235">US Virgin Islands</option>
  <option value="211">Uzbekistan</option>
  <option value="212">Vanuatu</option>
  <option value="213">Vatican City State</option>
  <option value="214">Venezuela</option>
  <option value="215">Vietnam</option>
  <option value="218">Wallis And Futuna Islands</option>
  <option value="271">Western Sahara</option>
  <option value="220">Yemen</option>
  <option value="224">Zambia</option>
  <option value="225">Zimbabwe</option>
</select>
```

**Comment by cachly** [comment]
**Original comment by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

Fixed issue #1078
